### PR TITLE
MAT-6992 error check when import test cases with special characters

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-6992](https://jira.cms.gov/browse/MAT-6992)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/cms/gov/madie/measure/services/TestCaseService.java [hunk 1]
```diff
@@ -765,17 +769,48 @@ protected void defaultTestCaseJsonForQdmMeasure(TestCase testCase, Measure measu
...
+                    testCaseImportRequest.getGivenNames() != null
+                        ? testCaseImportRequest.getGivenNames().get(0)
+                        : null)
+                .series(testCaseImportRequest.getFamilyName())
+                .build());
+      } catch (InvalidRequestException ex) {
+        return TestCaseImportOutcome.builder()
+            .patientId(testCaseImportRequest.getPatientId())
+            .successful(false)
+            .message("Test Case Title is required.")
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/services/TestCaseService.java]:
```suggestion
            .message(ex.getMessage())
```

File: src/main/java/cms/gov/madie/measure/services/TestCaseService.java [hunk 2]
```diff
@@ -765,17 +769,48 @@ protected void defaultTestCaseJsonForQdmMeasure(TestCase testCase, Measure measu
...
+            .build();
+      } catch (SpecialCharacterException ex) {
+        return TestCaseImportOutcome.builder()
+            .patientId(testCaseImportRequest.getPatientId())
+            .successful(false)
+            .message("Test Cases Group or Title cannot contain special characters.")
```
Comment (adongare) [inline src/main/java/cms/gov/madie/measure/services/TestCaseService.java:810]:
```suggestion
            .message(ex.getMessage())
```

Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/services/TestCaseService.java:810]: @adongare Thanks Ashok! Adopted the change.

Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/services/TestCaseService.java:810]: @adongare Fixed the test too. Thanks!

Comment (adongare) [reply src/main/java/cms/gov/madie/measure/services/TestCaseService.java:810]: thanks Cecilia! I think this is another similar case or maybe forgot to push the change?

Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/services/TestCaseService.java:810]: @adongare No for this exception, the message has to be Test Cases Group or Title cannot contain special characters. (based on the MAT-6992 requirement). The ex.getMessage will give: Test Case Group (or Title) can not have special characters followed by the list of the special characters, so they are not the same.

**Review by adongare — APPROVED** [review]

**Pull request merged**